Wikipedia/Code/WMFAuthTokenFetcher.swift

```diff
@@ -111,14 +112,17 @@ public class WMFAuthTokenFetcher: NSObject {
         let fetchToken = {
             self.fetchToken(ofType: context.tokenType, siteURL: context.siteURL, success: context.success, failure: context.failure)
         }
-        WMFAuthenticationManager.sharedInstance.loginWithSavedCredentials(context.siteURL, success: { (_) in
-            fetchToken()
-        }, userAlreadyLoggedInHandler: { (_) in
-            fetchToken()
-        }, failure: { (error) in
-            context.failure(error)
-        }) {
-            self.isRetryingLogin = false
+        DispatchQueue.main.async {
```

Comment:
This asyncs the `loginWithSavedCredentials` call, but not the callbacks. `self.isRetryingLogin = false` in the callbacks below could still be on any thread

Reply:
https://github.com/wikimedia/wikipedia-ios/pull/2650/commits/ecd79d425d82a99d0f480777233423a8f67a0e19 is this what you mean? 🧐

Reply:
@nambatee yes, but you can also delete the old `DispatchQueue.main.async` { above the entire method